mm2src/coins/utxo/rpc_clients/electrum_rpc/connection_managers/connection_context.rs

```diff
@@ -0,0 +1,60 @@
+use std::mem;
+use std::sync::{Arc, Mutex};
+
+use super::super::connection::ElectrumConnection;
+use super::super::constants::FIRST_SUSPEND_TIME;
+
+use common::now_ms;
+use keys::Address;
+
+/// A struct that encapsulates an Electrum connection and its information.
+#[derive(Debug)]
+pub struct ConnectionContext {
+    /// The electrum connection.
+    pub connection: Arc<ElectrumConnection>,
+    /// The list of addresses subscribed to the connection.
+    subs: Mutex<Vec<Address>>,
+    /// How long to suspend the server the next time it disconnects (in milliseconds).
+    next_suspend_time: Mutex<u64>,
+    /// When was the connection last disconnected.
+    disconnected_at: Mutex<u64>,
```

Comment:
Let's use [AtomicU64](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html) instead of `Mutex` here.

Reply:
I packaged the two things in a different type: https://github.com/KomodoPlatform/komodo-defi-framework/pull/1966/commits/0103f0b2ece3e34ab50c56378b891f8ab847082e

Or would using atomics be better perf-wise?

Reply:
Yes, atomics are significantly more reliable (as the load-lock can be customizable). You can still use them in the packed type.